src/spec-lang/ast/for_all.ts

```diff
@@ -54,19 +53,21 @@ export class SForAll extends SNode {
                 start === undefined && end === undefined,
                 `Can't specify both an array and start or end for forall.`
             );
-            this.start = new SNumber(bigInt(0), 10);
-            this.end = new SMemberAccess(array, "length");
+            // this.start = new SNumber(bigInt(0), 10);
+            // this.end = new SMemberAccess(container, "length");
         }
 
         this.expression = expression;
-        this.array = array;
+        this.container = container;
     }
 
     pp(): string {
-        if (this.array === undefined) {
-            return `(forall(${this.iteratorType.pp()} ${this.iteratorVariable.pp()} in ${this.start.pp()}...${this.end.pp()} ${this.expression.pp()}`;
+        if (this.container === undefined) {
+            return `(forall(${this.iteratorType.pp()} ${this.iteratorVariable.pp()} in ${(
+                this.start as SNode
+            ).pp()}...${(this.end as SNode).pp()} ${this.expression.pp()}`;
         } else {
-            return `(forall(${this.iteratorType.pp()} ${this.iteratorVariable.pp()} in ${this.array.pp()}) ${this.expression.pp()}`;
+            return `(forall(${this.iteratorType.pp()} ${this.iteratorVariable.pp()} in ${this.container.pp()}) ${this.expression.pp()}`;
```

Comment:
A case for early returns. Reduce nesting on `else`.

Reply:
Fixed in a97c7d9.